File: MobileApp/src/DTO/DefectDetailsDto.ts
import {DefectCategory} from '../enums/DefectCategory.ts';

export interface DefectDetailsDto {
  id: string;
  description: string;
  location: string;
  locationName: string;
  dateReported: Date;
  isFixed: boolean;
  defectCategory: DefectCategory;
  imageUrl: string;
}

File: Api/DTO/DefectDetailsDto.cs
using System.ComponentModel.DataAnnotations;
using Api.Enums;

namespace Api.DTO;

public class DefectDetailsDto
{
    [Required]
    public string Id { get; set; }
    [Required]
    public string Description { get; set; }
    [Required]
    public string Location { get; set; }
    [Required]
    public string LocationName { get; set; }
    [Required]
    public DateTime DateReported { get; set; }
    [Required]
    public bool IsFixed { get; set; }
    [Required]
    public DefectCategory DefectCategory { get; set; }
    [Required]
    public string ImageUrl { get; set; }
}